client/network/src/protocol/generic_proto/behaviour.rs

```diff
@@ -638,42 +710,86 @@ impl GenericProto {
 				*occ_entry.into_mut() = PeerState::Requested;
 			},
 
+			// Disabled (with non-expired ban) => DisabledPendingEnable
 			PeerState::Disabled {
-				open,
-				banned_until: Some(ref banned)
-			} if *banned > now => {
+				connections,
+				backoff_until: Some(ref backoff)
+			} if *backoff > now => {
 				let peer_id = occ_entry.key().clone();
-				debug!(target: "sub-libp2p", "PSM => Connect({:?}): But peer is banned until {:?}",
-					peer_id, banned);
+				debug!(target: "sub-libp2p", "PSM => Connect({:?}): But peer is backed-off until {:?}",
+					peer_id, backoff);
 
 				let delay_id = self.next_delay_id;
 				self.next_delay_id.0 += 1;
-				let delay = futures_timer::Delay::new(*banned - now);
+				let delay = futures_timer::Delay::new(*backoff - now);
 				self.delays.push(async move {
 					delay.await;
 					(delay_id, peer_id)
 				}.boxed());
 
 				*occ_entry.into_mut() = PeerState::DisabledPendingEnable {
-					open,
+					connections,
 					timer: delay_id,
-					timer_deadline: *banned,
+					timer_deadline: *backoff,
 				};
 			},
 
-			PeerState::Disabled { open, banned_until: _ } => {
-				debug!(target: "sub-libp2p", "PSM => Connect({:?}): Enabling connections.",
-					occ_entry.key());
-				debug!(target: "sub-libp2p", "Handler({:?}) <= Enable", occ_entry.key());
-				self.events.push_back(NetworkBehaviourAction::NotifyHandler {
-					peer_id: occ_entry.key().clone(),
-					handler: NotifyHandler::All,
-					event: NotifsHandlerIn::Enable,
-				});
-				*occ_entry.into_mut() = PeerState::Enabled { open };
+			// Disabled => Enabled
+			PeerState::Disabled { mut connections, backoff_until } => {
```

Comment:
```suggestion
			PeerState::Disabled { mut connections, backoff_until } => {
			debug_assert!(!connections.iter().any(|(_, s)| {
				matches!(s, ConnectionState::Open)
			}));			
```

Given that the back-off expired, one can assume that there are no open connections, right?

Reply:
Even if the back-off hasn't expired, there should never be any connection open in the `PeerState::Disabled` state.